### DISCUSSION TITLE
想了好几天，想的没睡着觉。callingPendingFunctors_会不会延迟回调？

### DISCUSSION THREAD

**Discussion by dxgzg** [chenshuo/muduo#688] [General]
我在想callingPendingFunctors_为false的时候，且loop为阻塞的时候，又不会wakeup，只能被迫等着下次事件触发，这样会不会导致延迟啊？

**Comment by chenshuo** [comment]
完全看不懂你想问什么。

你能不能用可以运行的代码复现你设想的场景？

**Comment by cellphonef** [comment]
> 我在想callingPendingFunctors_为false的时候，且loop为阻塞的时候，又不会wakeup，只能被迫等着下次事件触发，这样会不会导致延迟啊？

我认为functors只会在使用queueInLoop或runInLoop时进行添加，如果不调用这两个函数的话就不会有functor需要执行也自然不存在延迟的问题，而调用这两个函数在必要时会进行唤醒，因此也能及时执行functor。

**Comment by TalentFang** [comment]
此时，如果是本线程会直接执行函数，如果是外部线程会调用wakeup吧
